Dockerfile

```diff
@@ -12,16 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.13-buster AS build
-WORKDIR /go/src/github.com/vmware-tanzu/velero-plugin-for-aws
-# copy vendor in separately so the layer can be cached if the contents don't change
-COPY vendor vendor
-COPY velero-plugin-for-aws velero-plugin-for-aws
+# golang:1.13.2-buster
+# https://hub.docker.com/layers/golang/library/golang/1.13.2-buster/images/sha256-36c384d8593f8989b7a50bebf9569c9f6a43a6881baa8a248a84251d2ca17ddb
+ARG GOLANG_IMAGE_REF=sha256:36c384d8593f8989b7a50bebf9569c9f6a43a6881baa8a248a84251d2ca17ddb
+
+# debian:buster-20191224
+# https://github.com/docker-library/official-images/commit/1b4809bba555e354a6ea13b01cfb0fcca705f7ac#diff-55350c691707bcc61370b117bb4d37d9R44
+ARG DEBIAN_IMAGE_REF=sha256:e4c1417236abc57971755ca2bfccd546cbca45b33daf66001a5addae4bf78517
+
+FROM golang:1.13.2-buster@${GOLANG_IMAGE_REF} as golang
+ARG VELERO_GIT_REPO="github.com/vmware-tanzu/velero-plugin-for-aws"
+
+RUN mkdir -p "/go/src/${VELERO_GIT_REPO}"
+
+COPY . "/go/src/${VELERO_GIT_REPO}"
+
+
+WORKDIR /go/src/${VELERO_GIT_REPO}
 RUN CGO_ENABLED=0 GOOS=linux go build -v -o /go/bin/velero-plugin-for-aws ./velero-plugin-for-aws
 
 
-FROM ubuntu:bionic
+FROM debian:buster-slim@${DEBIAN_IMAGE_REF}
```

Comment:
why are we switching the base image for this and what side effects can this have?

Reply:
So, this is an opinionated decision. 
I used debian for a few reasons:
- I prefer running headless servers on debian rather than ubuntu. It's just a habit, I have no technical arguments at the moment for it, except I feel like debian is 'more open' than canonical/ubuntu.
- I had the build logic handy and the debian build digest.

Truthfully though, I don't think it actually matters for the end-user. The real initiative here was to change the group to `group`.

This could probably be a smaller image like alpine or busybox. The entrypoint requires `bash` and `cp`--but really it just requires `cp` to copy the files to a path mounted by the `velero` server.